pkg/apis/triggers/v1alpha1/trigger_types.go

```diff
@@ -39,9 +39,12 @@ type TriggerSpec struct {
 }
 
 type TriggerSpecTemplate struct {
-	Ref        *string              `json:"ref,omitempty"`
-	APIVersion string               `json:"apiversion,omitempty"`
-	Spec       *TriggerTemplateSpec `json:"spec,omitempty"`
+	// Deprecated: Use Ref instead
+	// To be removed in a later release #911
+	DeprecatedName string               `json:"name,omitempty"`
```

Comment:
Not blocking just wondering:
Why does the `omitempty` fix the problem described in slack?

Reply:
Because omitempty ensures that when this struct is store, it stores nothing related to DeprecatedName field but Ref using defaults. Previously, default wasn't working properly because it stored "".

Reply:
What I mean is: We have removed the name field completely already, creating a new CR should therefore not have the field anymore. So even if it was stored with "" we should fix anything going wrong in that case.

Readding the field here kind of implies that users first need to upgrade to this specific version before updating to the next minor.

Reply:
I think we can recommend the user to directly upgrade to v0.11.0 from v0.10.2

Reply:
So then it would make sense to put this change into `v0.10.3` instead of `v0.11.1`, right?

Reply:
yeah doing a 0.10.3 is also an option. Though that would mean users would first have to move to v0.10.3 and then to v0.11.0/v0.11.1